### ISSUE TITLE
Bug Report:  AMX metrics does not work in cluster nor deployment group / FISH-5658

### ISSUE THREAD

**Issue by sgflt** [payara/Payara#5249]
# Description #
----------
If we enable AMX and configure metrics.xml in domain/config folder. In cluster/dg only single instance have metrics propagated correctly. Other instances throws exceptions. 

<details>

```
On instance test2:
java.lang.IllegalStateException: javax.management.InstanceNotFoundException: amx:pp=/mon/server-mon[test1],type=connection-queue-mon,name=network//connection-queue
```
</details>

## Expected Outcome ##
All configured metrics are visoble on all instances.

## Current Outcome ##
Only the first instance matching the name of the first metric works.

## Steps to reproduce ## 

1. create cluster or deployment group with at least two instances
2. place metrics.xml to domain/config
3. set monitoring levels to high
4. access /metrics endpoint on all instances (only single one works)

metrics.xml
<details>

```
<config><vendor>
    <metadata>
      <name>http.connection.queue.http.numconnopen</name>
      <mbean>amx:pp=/mon/server-mon[test1],type=connection-queue-mon,name=network//connection-queue/countopenconnections#count</mbean>
      <type>gauge</type>
      <unit>none</unit>
      <displayName>numconnopenPool</displayName>
      <description>The number of open/active http connections</description>
    </metadata>

    <metadata>
      <name>http.connection.queue.http.numconnopen</name>
      <mbean>amx:pp=/mon/server-mon[test2],type=connection-queue-mon,name=network//connection-queue/countopenconnections#count</mbean>
      <type>gauge</type>
      <unit>none</unit>
      <displayName>numconnopenPool</displayName>
      <description>The number of open/active http connections</description>
    </metadata>

</vendor></config>
```
</details>

## Environment ##

- **Distribution**: Full 
- **JDK Version**: OpenJDK 11 
- **Operating System**: Fedora 34

**Comment by patnbg** [comment]
Defining the metric only once using %s as a placeholder in the mbean name might fix the problem. 
Something like that:


```
<config>
  <vendor>
      <metadata>
        <tags>
          <tag>
              <name>name</name>
              <value>%s</value>
          </tag>
        </tags>
        <name>http.connection.queue.http.numconnopen</name>
        <mbean>amx:pp=/mon/server-mon[%s],type=connection-queue-mon,name=network//connection-queue/countopenconnections#count</mbean>
        <type>gauge</type>
        <unit>none</unit>
        <displayName>numconnopenPool</displayName>
        <description>The number of open/active http connections</description>
      </metadata>
  </vendor>
</config>
```

**Comment by sgflt** [comment]
It does not seem to be general solution. It works for connection-queue but produces nonsense for jdbc:
`vendor_jdbc_connection_pool_numconnused{instance="/mon/server-mon[test1]",pool="/mon/server-mon[test1]"} 0`

Used xml:
<details>

```
<config><vendor>    <metadata>
      <tags>
        <tag>
          <name>instance</name>
          <value>%s</value>
        </tag>
        <tag>
          <name>pool</name>
          <value>%s</value>
        </tag>
      </tags>
     <name>jdbc.connection.pool.numconnused</name>
      <mbean>amx:pp=/mon/server-mon[%s],type=jdbc-connection-pool-mon,name=resources/%s/numconnused#current</mbean>
      <type>gauge</type>
      <unit>none</unit>
      <displayName>numconnusedpool</displayName>
      <description>The total number of already used connections.</description>
    </metadata>

    <metadata>
        <tags>
          <tag>
              <name>name</name>
              <value>%s</value>
          </tag>
        </tags>
        <name>http.connection.queue.http.numconnopen</name>
        <mbean>amx:pp=/mon/server-mon[%s],type=connection-queue-mon,name=network//connection-queue/countopenconnections#count</mbean>
        <type>gauge</type>
        <unit>none</unit>
        <displayName>numconnopenPool</displayName>
        <description>The number of open/active http connections</description>
      </metadata>
</vendor></config>
```
</details>

**Comment by rdebusscher** [comment]
Fixed in version 5.2021.8. 

There is now an option to set instance name wildcard in the metrics.xml file. https://docs.payara.fish/community/docs/documentation/microprofile/metrics/vendor-metrics.html#instance-names
